Core/HLE/sceFont.cpp

```diff
@@ -601,6 +735,36 @@ void PostOpenCallback::run(MipsCall &call) {
 	fontLib->SetFileFontHandle(v0);
 }
 
+void PostOpenAllocCallback::run(MipsCall &call) {
+	FontLib *fontLib = fontLibList[fontLibID_];
+	u32 v0 = currentMIPS->r[MIPS_REG_V0];
+	fontLib->SetOpenAllocatedAddress(fontIndex_, v0);
+	call.setReturnValue(fontHandle_);
+}
+
+void PostCharInfoAllocCallback::run(MipsCall &call) {
+	FontLib *fontLib = fontLibList[fontLibID_];
+	u32 v0 = currentMIPS->r[MIPS_REG_V0];
+	if (v0 == 0) {
+		call.setReturnValue(ERROR_FONT_OUT_OF_MEMORY); // From JPCSP, if alloc size is 0, still this error value?
+	} else {
+		fontLib->SetCharInfoBitmapAddress(v0);
+		call.setReturnValue(0);
+	}
+}
+
+void PostCharInfoFreeCallback::run(MipsCall &call) {
+	FontLib *fontLib = fontLibList[fontLibID_];
+	fontLib->SetCharInfoBitmapAddress(0);
+
+	u32 allocSize = charInfo_->bitmapWidth * charInfo_->bitmapHeight;
+	PostCharInfoAllocCallback *action = (PostCharInfoAllocCallback *)__KernelCreateAction(actionPostCharInfoAllocCallback);
+	action->SetFontLib(fontLibID_);
+
+	u32 args[2] = { fontLib->userDataAddr(), allocSize };
+	__KernelDirectMipsCall(fontLib->allocFuncAddr(), action, args, 2, true);
```

Comment:
@unknownbrackets , Alloc for the new char after free old here, matches the second case you point out, I mean:
sceFontGetCharInfo('A'); // calls allocate, keeps A allocated
sceFontGetCharInfo('B'); // calls free on A, calls alloc on B
sceFontGetCharInfo('C'); // calls free on B, calls alloc on C

Reply:
Oh, I thought it just alternated.  I missed indeed that this also allocates.

-[Unknown]